automates/model_analysis/graph_manipulation.py

```diff
@@ -0,0 +1,634 @@
+                visited = [False] * visited_size
+                visited[0:visited_size_old] = copy.deepcopy(visited_old)
+                visited_names = [None] * visited_size
+                visited_names[0:visited_size_old] = copy.deepcopy(visited_names_old)
+            visited[visited_top - 1] = el
+            visited_names[visited_top - 1] = el_name
+            el_name_in_z = el_name in z
+            if el and (not el_name_in_z):
+                visitable_parents = list((set(parents_unsort([el_name], g)) - set([el_name])) & set(an_xyz))
+                visitable_children = list((set(children_unsort([el_name], g)) - set([el_name])) & set(an_xyz))
+                n_vis_pa = len(visitable_parents)
+                n_vis_ch = len(visitable_children)
+                if n_vis_pa + n_vis_ch > 0:
+                    while n_vis_pa + n_vis_ch + stack_top > stack_size:
+                        stack_old = copy.deepcopy(stack)
+                        stack_names_old = copy.deepcopy(stack_names)
+                        stack_size_old = stack_size
+                        stack_size = 2 * stack_size
+                        stack = [False] * stack_size
+                        stack[0:stack_size_old] = copy.deepcopy(stack_old)
+                        stack_names = [None] * stack_size
+                        stack_names[0:stack_size_old] = copy.deepcopy(stack_names_old)
+                    stack_add = stack_top + n_vis_pa + n_vis_ch
+                    stack[stack_top:stack_add] = [True] * n_vis_pa + [False] * n_vis_ch
+                    stack_names[stack_top:stack_add] = copy.deepcopy(visitable_parents) + copy.deepcopy(
+                        visitable_children)
+                    stack_top = stack_add
+            elif not el:
+                if not el_name_in_z:
+                    visitable_children = list((set(children_unsort(el_name, g)) - set(el_name)) & set(an_xyz))
+                    n_vis_ch = len(visitable_children)
+                    if n_vis_ch > 0:
+                        while n_vis_ch + stack_top > stack_size:
+                            stack_old = copy.deepcopy(stack)
+                            stack_size_old = stack_size
+                            stack_names_old = copy.deepcopy(stack_names)
+                            stack_size = 2 * stack_size
+                            stack = [False] * stack_size
+                            stack[0:stack_size_old] = copy.deepcopy(stack_old)
+                            stack_names[0:stack_size_old] = copy.deepcopy(stack_names_old)
+                        stack_add = stack_top + n_vis_ch
+                        stack[stack_top:stack_add] = [False] * n_vis_ch
+                        stack_names[stack_top:stack_add] = copy.deepcopy(visitable_children)
+                        stack_top = stack_add
+                if el_name in an_z:
+                    visitable_parents = list((set(parents_unsort([el_name], g)) - set([el_name]) & set(an_xyz)))
```

Comment:
@aswinchester see my comment above and try to handle this block in that function as well.

Reply:
(see above)